### PR TITLE
Copying over https://web.dev/app-history-api/

### PR DESCRIPTION
**Pull request by jakearchibald**
This PR copies over the old article from https://web.dev/app-history-api/. It includes changes from https://github.com/GoogleChrome/developer.chrome.com/pull/2897.

### CODE REVIEW

**Comment by jakearchibald** [comment]
@rachelandrew this is the article you just reviewed. I split it into two PRs so one was copying the old doc over, and the other was just the changes to the old doc. This PR is now a combination of both, so it can be merged into main.

File: site/_data/docs/web-platform/toc.yml [hunk 1]
```diff
@@ -2,6 +2,8 @@
 - url: /docs/web-platform/origin-trial-troubleshooting
 
 - title: i18n.docs.web-platform.shipped
+  sections:
+    - url: /docs/web-platform/navigation-api
```
Comment (jakearchibald) [inline site/_data/docs/web-platform/toc.yml:6]: @rachelandrew this is the only change from what you've seen already.

File: site/_data/i18n/authors.yaml [hunk 2]
```diff
@@ -825,3 +825,8 @@ egsweeny:
     en: Elizabeth Sweeny
   description:
     en: Elizabeth is a product manager working on developer tooling for Chrome Web Platform.
+samthor:
+  title:
+    en: Sam Thorogoodd
```
Comment (rachelandrew) [inline site/_data/i18n/authors.yaml:830]:
```suggestion
    en: Sam Thorogood
```

**Review by rachelandrew — APPROVED** [review]

**Comment by rachelandrew** [comment]
@jakearchibald I'll merge this and once it is showing up on the site I'll do the redirects.

**Pull request merged**